.github/actions/query_issues/action.yml

```diff
@@ -0,0 +1,40 @@
+name: Query issues using GraphQL API
+inputs:
+  token:
+    description: 'The GITHUB_TOKEN secret'
+    required: true
+  query:
+    description: "The GraphQL search query to be executed"
+    required: true
+    type: string
+outputs:
+  query_result:
+    description: "The result of the query"
+    value: ${{ steps.get_issues.outputs.query_result }}
+
+runs:
+  using: "composite"
+  steps:
+    - id: get_issues
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.token }}
+      run: |
+        result="$(gh api graphql \
+          --field search_query="${{ inputs.query }}" \
+          --raw-field query='
+            query($search_query: String!, $cursor: String) {
+              search(first: 100, query: $search_query, type: ISSUE, after: $cursor) {
```

Comment:
100 might not be right for all situations. I think this should be a parameter of the action, with 100 simply being the default.

Reply:
This is only the maximum number of results to be returned by the graphql query. It will not necessarily be the number of issues the action will comment on. We can limit that to the `stale-issue.yml` instead.

And the pagination handle is also missing here, so I'd need to handle the `cursor` properly and do repeated queries until all results are obtained if we want the action to capture all issues that match the filter.